repository/sqlite_repository.py

```diff
@@ -296,57 +306,77 @@ async def read_comments(
 
-        if rating_max is not None:
-            query.append(Comment.rating <= rating_max)
+            if rating_max is not None:
+                query.append(Comment.rating <= rating_max)
 
-        if content_search is None:
-            if len(query) == 0:
-                return await self.get_all_comments()
-            else:
-                query = await Comment.filter(*query)
-                return list(query)
-
-        if content_search is not None:
-            table = Comment.get_table()
-            with Session(Comment.__metadata__.database.engine) as session:
-                result = (
-                    session.query(table)
-                    .where(table.c.comment.regexp_match(content_search))
-                    .all()
-                )
-            comments_searched: List[Comment] = Comment.parse_results(result, [], False)
-            if len(query) == 0:
-                return comments_searched
+            if content_search is None:
+                if len(query) == 0:
+                    comments = await self.get_all_comments()
+                else:
+                    query = await Comment.filter(*query)
+                    comments = list(query)
             else:
-                query = await Comment.filter(*query)
-                query_comments = list(query)
-                common_comments = [x for x in comments_searched if x in query_comments]
-                return common_comments
+                table = Comment.get_table()
+                with Session(Comment.__metadata__.database.engine) as session:
+                    result = (
+                        session.query(table)
+                        .where(table.c.comment.regexp_match(content_search))
+                        .all()
+                    )
+                comments_searched: List[Comment] = Comment.parse_results(result, [], False)
+                if len(query) == 0:
+                    comments = comments_searched
+                else:
+                    query = await Comment.filter(*query)
+                    query_comments = list(query)
+                    comments = [x for x in comments_searched if x in query_comments]
+
+        # Perform pagination
+        total_comments = len(comments)
+        total_pages = ceil(total_comments / page_size)
+        start_index = (page - 1) * page_size
+        end_index = start_index + page_size
+        paginated_comments = comments[start_index:end_index]
+    
+
+        # Prepare pagination information
+        next_page = f"/comments?page={page + 1}&pageSize={page_size}" if page < total_pages else None
+        prev_page = f"/comments?page={page - 1}&pageSize={page_size}" if page > 1 else None
+
+        return {
+            "results": paginated_comments,
+            "page": page,
+            "page_size": page_size,
+            "total": total_comments,
+            "next_page": next_page,
+            "prev_page": prev_page,
+        }
```

Comment:
Move the pagination logic to the endpoint method. It's an API feature, not a database feature